hack/ipa_debug_tools.sh

```diff
@@ -0,0 +1,227 @@
+        printf "${BLUE}%-40s %-22s${NC}\n" "➕ ${name}" "$(numfmt --to=iec "${bsize}")"
+    done
+
+    echo "======================================================================================================="
+
+    # List files only in dir2
+    echo -e "\n${GREEN}🎯 Only in Master IPA (${dir2}):${NC}"
+    echo "-------------------------------------------------------------------------------------------------------"
+    comm -13 /tmp/dir1_list /tmp/dir2_list | while IFS= read -r name; do
+        isize=$(du -sb "${dir2}/${name}" 2>/dev/null | cut -f1)
+        echo -e "${isize}\t${name}"
+    done | sort -k1,1nr | while IFS=$'\t' read -r isize name; do
+        printf "${GREEN}%-40s %-22s${NC}\n" "➕ ${name}" "$(numfmt --to=iec "${isize}")"
+    done
+
+    # Print some empty lines for clarity
+    echo ""
+    echo ""
+    echo ""
+    
+    # Remove temporary dir lists
+    rm /tmp/dir1_list /tmp/dir2_list
+}
+
+# Print a rpm package list. The print omits packages that are identical in both
+#the built ipa and master.
+# Check if packages are a part of CentOS Stream in https://pkgs.org
+compare_rpm_packages() {
+    basedir1="${1:-/tmp/debug-initramfs/build-ipa-initramfs}" 
+    basedir2="${2:-/tmp/debug-initramfs/master-ipa-initramfs}"
+    compfile1="/tmp/build-ipa-rpm.txt"
+    compfile2="/tmp/master-ipa-rpm.txt"
+
+    sudo chroot "${basedir1}" rpm -qa | sort > "${compfile1}"
+    sudo chroot "${basedir2}" rpm -qa | sort > "${compfile2}"
+
+    common_count=$(comm -12 "${compfile1}" "${compfile2}" | wc -l)
+    echo -e "\nNumber of common packages: ${common_count}\n"
+    # Extract base package names and join for fuzzy matching
+    awk -F'-[0-9]' '{print $1}' "${compfile1}" | sort > /tmp/build-ipa-base.txt
+    awk -F'-[0-9]' '{print $1}' "${compfile2}" | sort > /tmp/master-ipa-base.txt
+
+    printf "%-65s | %-65s\n" "In ${basedir1}" "In ${basedir2}"
+    printf -- "----------------------------------------------------------------------------------------------------------------------------------\n"
+
+    join -a1 -a2 -e '' -o 1.1,2.1 /tmp/build-ipa-base.txt /tmp/master-ipa-base.txt | \
+    while IFS=' ' read -r left right; do
+        # Find the full package lines for each base name
+        lfull=$(grep -m1 "^${left}-" "${compfile1}" || echo "")
+        rfull=$(grep -m1 "^${right}-" "${compfile2}" || echo "")
+        # Skip printing if both lines are non-empty and exactly the same
+        if [[ -n "${lfull}" && "${lfull}" == "${rfull}" ]]; then
+            continue
+        fi
+        printf "%-65s | %-65s\n" "${lfull}" "${rfull}"
+    done
+
+    sudo rm -f "${compfile1}" "${compfile2}" /tmp/build-ipa-base.txt /tmp/master-ipa-base.txt
+}
+
+# After debugging and comparison operations, temporary directories can be cleaned up.
+# Usage:
+#   ./ipa_debug_tools.sh clean_up_debug_dirs
+# -----------------------------------------------------------------------------
+clean_up_debug_dirs() {
```

Comment:
Cleaning up all the temp files should be done here, and the cleanup should be trapped function so it gets called even if we exit mid run.